### PR TITLE
change softmax cross entropy hessian

### PR DESCRIPTION
**Pull request by gklab**
Signed-off-by: gklab [sanxianwei@163.com](mailto:sanxianwei@163.com)
Changes:

1. change compute the hessian of softmax cross entropy

before
```
y_pred * (1 - y_pred)
```
after
```
2 * y_pred * (1 - y_pred)
```

you can refer to this link for the reason:
https://github.com/dmlc/xgboost/issues/1825

### CODE REVIEW

**Comment by talkingwallace** [comment]
Hi, our computation of Hession follows Xgboost& Lightgbm, see:
2023-04-25 18-58-56屏幕截图
Could you please tell us why '2*' is needed here?
Thanks.

**Comment by gklab** [comment]
> Hi, our computation of Hession follows Xgboost& Lightgbm, see: 2023-04-25 18-58-56屏幕截图 Could you please tell us why '2*' is needed here? Thanks.

multiclass hessian code:
image

The reason is that the Taylor formula of the second derivative may not be optimal in some cases.
image

**Comment by talkingwallace** [comment]
Hi! Thank you so much for you PR! We will update this modification in our next version. :D

**Review by dylan-fan — APPROVED** [review]

**Review by mgqa34 — APPROVED** [review]
